### PR TITLE
Limited progress made building for Android (and iOS)

### PR DESCRIPTION
**Pull request by jhugman**
I am vaguely following along with [this post from Oscar Franco](https://ospfranco.com/post/2024/05/08/react-native-rust-module-guide/).

My goal is to generate typescript and JSI/C++ that talks to Rust.

I have this running in vanilla hermes for the sample library "rondpoint" I'm using here.

Now I'm trying to build this for Android:

- use Codegen to make a turbomodule for a simple `installRustCrate`/`cleanupRustCrate` lifecycle.
- At the end of the Codegen'd Objective-C/JNI, find the `jsi::Runtime` and call into a C++ method to `NativeRonpoint::registerModule`

I'm able to demonstrate that the Codegen piece, finding the `jsi::Runtime` and calling into common C++ works.

The `generate.sh` essentially does:

```sh
git clone https://github.com/jhugman/uniffi-bindgen-react-native rust_modules/uniffi-bindgen-react-native
```
to checkout the git repo

then for each target, we use `cargo ndk` to build the rust library:

```sh
cargo ndk \
    --target some-android-linux-target \
    --platform 21 \
    --manifest-path fixtures/ronpoint/Cargo.toml \
    --output-dir android/src/main/jniLibs \
    build
```

These produce `.so` files, and copy them into the android jniLibs directory.

It then uses the `uniffi-bindgen-react-native` generator to generate typescript and cpp from the so library,
and puts the files in `cpp/generated` and `src/generated`.

Then when `yarn example start` is run with android, then the `CMakeLists.txt`
is executed.

```cmake
cmake_path(
  SET MY_SDK_LIB
  ${CMAKE_SOURCE_DIR}/src/main/jniLibs/${ANDROID_ABI}/libuniffi_rondpoint.so
  NORMALIZE
)
```

based on Oscar's:
```cmake
make_path(SET MY_SDK_LIB ${CMAKE_CURRENT_SOURCE_DIR}/jniLibs/${ANDROID_ABI}/libmy_sdk.a NORMALIZE)
add_library(my_sdk STATIC IMPORTED)
set_target_properties(my_sdk PROPERTIES IMPORTED_LOCATION ${MY_SDK_LIB})
```

**But notice** that:
- `libmy_sdk.a` is a staticlib, not a shared object file; and `cargo ndk`
- `make_path` doesn't exist, but I think I should be using `cmake_path`

What I think I want on Android is for to make a `.so` with C++ which calls into the Rust `.so` file, so it
can be loaded with `System.loadLibrary("matrix-rust-sdk");`.

I added a `loadLibrary("uniffi_rondpoint")` because it was failing to get the right relative path from
`matrix-rust-sdk`.

Even then though it was failing at runtime; the following is seen in `adb logcat`:

```sh
E ReactNativeJS: TypeError: _rondpointFfi.default.uniffi_uni(...)point_fn_func_copie_dictionnaire is not a function (it is undefined)
E ReactNativeJS: Invariant Violation: "MatrixRustSdkExample" has not been registered. This can happen if:
E ReactNativeJS: * Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
E ReactNativeJS: * A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called.
```

Having added logging, I don't see anything in the log that I have put there, which would suggest:
either the logging from C++ and Javascript isn't setup properly OR it's not getting to those lines.

In summary:

- I have Typescript --> Codegen --> Java/JNI --> C++ working on Android
- I have Typescript --> Codegen --> Objective-C --> C++ working on iOS
- I have Typescript --> C++ --> Rust working on Desktop

I need help with 

the build steps for Android and iOS.

### CODE REVIEW

**Review by zzorba — APPROVED** [review]
Android build looks good
